### ISSUE TITLE
TRY301 disregards whether the raised exeption type is actually caught or not

### ISSUE THREAD

**Issue by runfalk** [astral-sh/ruff#5246]
I believe the following is a false positive:

```python
# example.py
try:
    raise ValueError("foo")
except TypeError:
    print("We don't catch ValueError")
```

```
$ poetry run ruff example.py 
example.py:3:5: TRY301 Abstract `raise` to an inner function
Found 1 errors.
```

I would expect `TRY301` only if we actually caught the `ValueError`.

## Relevant metadata
```
$ poetry run python --version
Python 3.11.3
```

```
$ poetry run ruff --version
ruff 0.0.274
```

```
[tool.ruff]
target-version = "py311"

select = [
    # ...
    "TRY",  # tryceratops
    # ...
]
ignore = [
    # ...

    # tryceratops
    # (these options are probably irrelevant)
    # Allow long inline exception messages
    "TRY003",
    # Allow return statements in try blocks
    "TRY300",
]
```

**Comment by evanrittenhouse** [comment]
You can assign this to me
